src/app.test.ts

```diff
@@ -1,51 +1,267 @@
-import { test, beforeEach, afterEach } from 'tap'
+import { afterEach, before, beforeEach, teardown, test } from 'tap'
 import tmp from 'tmp'
+import path from 'path'
+import fs from 'fs'
+import { FastifyInstance } from 'fastify'
+import DB, { Database as DatabaseInstance } from 'better-sqlite3'
 
 import app from './app'
 import mapboxRasterTilejson from './fixtures/good-tilejson/mapbox_raster_tilejson.json'
-import { getTilesetId } from './lib/tilestore'
+import { getTilesetId } from './lib/utils'
+import { TileJSON, validateTileJSON } from './lib/tilejson'
+import { server as mockTileServer } from './mocks/server'
+
+import { IdResource } from './api'
 
 tmp.setGracefulCleanup()
 
-beforeEach((done, t) => {
+type TestContext = {
+  accessDb: (cb: (db: DatabaseInstance) => void) => void
+  server: FastifyInstance
+  sampleTileJSON: TileJSON
+}
+
+function assertSampleTileJSONIsValid(data: unknown): asserts data is TileJSON {
+  if (!validateTileJSON(data)) {
+    const message = `Sample input does not conform to TileJSON schema spec: ${JSON.stringify(
+      validateTileJSON.errors,
+      null,
+      2
+    )}`
+
+    throw new Error(message)
+  }
+}
+
+before(() => {
+  // Check if prisma/migrations directory exists in project
+  if (!fs.existsSync(path.resolve(__dirname, '../prisma/migrations'))) {
+    throw new Error(
+      'Could not find prisma migrations directory. Make sure you run `npm run prisma:migrate-dev -- --name MIGRATION_NAME_HERE` first!'
+    )
+  }
+
+  assertSampleTileJSONIsValid(mapboxRasterTilejson)
+
+  mockTileServer.listen()
+})
+
+beforeEach((t) => {
   const { name: dataDir } = tmp.dirSync({ unsafeCleanup: true })
-  t.context.app = app({ logger: false }, { dataDir })
-  done()
+
+  const dbPath = path.resolve(dataDir, 'test.db')
```

Comment:
We should create this in a temp dir, and delete it after each test.

Reply:
the `dataDir` comes from the tmp call so it's technically created within it. does that function differently than what you're suggesting?

Reply:
ah I missed the line above, my bad!